web_src/js/features/repo-settings-webhook.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * Webhook settings functionality
+ */
+
+function toggleLimitField(fieldName: string, enabled: boolean): void {
+  const field = document.querySelector<HTMLInputElement>(`input[name="${fieldName}"]`);
+  if (field) {
+    field.disabled = !enabled;
+  }
+}
+
+export function initRepoSettingsWebhook(): void {
```

Comment:
The logic is almost the same as the code touched in "Add toggleClass function in dom.ts (#34063)": use checkbox or radio input to toggle some elements, enable some while disable others .